### DISCUSSION TITLE
backward compatibility with XML-RPC?

### DISCUSSION THREAD

**Discussion by zEdS15B3GCwq** [jesec/rtorrent#17] [General]
Hi,

Do you intend to keep this version of rtorrent backward compatible with other UIs (in particular ruTorrent) that use XML-RPC? I have been using both ruTorrent and Flood to access jesec/rtorrent, which worked up until I pulled the latest image from a few days(?) ago. Now I'm getting JSON parse errors in the replies to ruTorrent's attempts to contact rtorrent; Flood works fine. I am guessing that jesec/rtorrent is moving on to JSON-based rpc, but I was wondering if this is an intended drop of old functionality or a bug. For now, I have reverted to the stock 0.9.8 rtorrent, which works with both UIs.

**Comment by jesec** [comment]
Um. Interesting. XMLRPC is still supported.

**Reply by zEdS15B3GCwq** [reply]
Okies, thanks for the quick response. I guess I'll recreate the error and report back.

**Reply by zEdS15B3GCwq** [reply]
A normal exchange of xmlrpc messages logged from ruTorrent with stock 0.9.8 rtorrent:

```
[ruTorrent -> rtorrent] <?xml version="1.0" encoding="UTF-8"?><methodCall><methodName>system.multicall</methodName>...
[ruTorrent <- rtorrent] Status: 200 OK
Content-Type: text/xml
Content-Length: 485

<?xml version="1.0" encoding="UTF-8"?>
...
```

With the executable from the latest jesec/rtorrent:

```
[ruTorrent -> jesec/rtorrent] <?xml version="1.0" encoding="UTF-8"?><methodCall><methodName>system.client_version</methodName><params>
</params></methodCall>
[(same but actual sent message contents)] 26:CONTENT_LENGTH128SCGI1,<?xml version="1.0" encoding="UTF-8"?><methodCall><methodName>system.client_version</methodName><params>
</params></methodCall>
[ruTorrent <- jsesec/rtorrent] Status: 200 OK
Content-Type: application/json
Content-Length: 209

{"error":{"code":-32700,"message":"parse error: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: '<'"},"id":null,"jsonrpc":"2.0"}

```

As ruTorrent seems to require that rtorrent itself has access to PHP, I load the contents of the jesec/rtorrent docker container into a php:7-alpine image. This worked so far.

**Reply by jesec** [reply]
Please check out the latest rolling build when it is ready.

**Reply by zEdS15B3GCwq** [reply]
~~The issue is gone on 0.9.8-r14. Thanks for fixing it.~~

Edit:

Spoke too early. Controlling the torrents from the ruTorrent interface works OK, but I get these errors when I press the 'settings' button:

```
[2021-06-15 13:34:52] XMLRPC Error: Method 'network.listen.is_open' not defined [getsettings]
[2021-06-15 13:34:52] XMLRPC Error: Method 'network.port.randomize' not defined [getsettings]
[2021-06-15 13:34:52] XMLRPC Error: Method 'network.port.range' not defined [getsettings]
```

Also - though this may or may not be related to your version of rtorrent, - there's another error message when loading ruTorrent:

```
ruTorrent cannot determine the UID of rTorrent user. Some functionality will be unavailable.
```

This has been reported elsewhere for `ruTorrent <3.10 + rTorrent >=0.9.8` and the solution was to upgrade to `ruTorrent 3.10 (beta)` [[1](https://github.com/linuxserver/docker-rutorrent/issues/149), [2](https://github.com/linuxserver/docker-rutorrent/issues/149)]. However, I'm already using release `3.10` and still getting it so it may be relevant?

Edit2:

BtW, the _vanilla_ version of rtorrent I'm using without issues is `0.9.8-r1` (latest) apk for Alpine.
